utils/decode_kallsyms.py

```diff
@@ -0,0 +1,53 @@
+import bisect
+import re
+
+
+def kallsyms_search(kallsyms_lines: list[str], address: int):
+    # kallsyms lines looks like this
+    #
+    # ffffffffaa88e2f0 T kfree
+    # ffffffffaa88e400 T __ksize
+    # ffffffffaa88e530 T ksize
+    key = lambda line: int(line.split()[0], 16) - 1
+    return bisect.bisect_right(kallsyms_lines, address, key=key) - 1
+
+
+def cut_addresses_from_log_line_kernel_callstack(line):
+    my_regex = re.compile('0x([0-9a-fA-F]{16})')
+    return my_regex.findall(line)
+
+
+def get_lines_from_log(file_path):
+    file_content = {}
+    with open(file_path, 'r') as file:  # pylint: disable=unspecified-encoding
+        for line in file:
+            if ('kernel callstack' in line and '0x' in line) or 'Reactor stalled for' in line:
+                file_content[line] = cut_addresses_from_log_line_kernel_callstack(line)
+    return file_content
+
+
+def get_kallsyms(path_to_kallsyms):
+    with open(path_to_kallsyms, 'r') as file:  # pylint: disable=unspecified-encoding
+        file_content = file.readlines()
+    return file_content
+
+
+def append_content_to_file(file_stream, content):
+    file_stream.write(content + '\n')
+
+
+def decode_kernel_callstacks(results_file='results.log', input_file='system.log', kallsyms_file='kallsyms',
```

Comment:
I would be nice if there would be an example showing how to use it.

i.e. now you can run this from commandline, so you need to write more code to use it.

Reply:
it can be called from a simple Python code, and give the relevant files, and collect the output...
the idea of this is to futurely allow us to automatically decode kernel callstacks from within SCT, and not a command line script (that can be easily adapted from this existing code BTW)

Reply:
event something like, probably would do the trick :

```python
if __name__ == "__main__":
      decode_kernel_callstacks()
```

Reply:
i can be added later, because it don't think it should be part of me adding this tool to SCT repo, since it is not yet used anywhere

Reply:
I was more concern about someone that needs it manually now, and will start to figure out himself what exactly he need todo.

later automating it, is a different story